dev/TEAL.md

```diff
@@ -242,6 +292,7 @@ Some of these have immediate data in the byte or bytes after the opcode.
 | 53 | GlobalNumByteSlice | uint64 | Number of global state byteslices in ApplicationCall. LogicSigVersion >= 3. |
 | 54 | LocalNumUint | uint64 | Number of local state integers in ApplicationCall. LogicSigVersion >= 3. |
 | 55 | LocalNumByteSlice | uint64 | Number of local state byteslices in ApplicationCall. LogicSigVersion >= 3. |
+| 56 | ExtraProgramPages | uint64 | Number of additional pages for each of the application's approval and clear state programs. An ExtraProgramPages of 1 means 2048 more total bytes, or 1024 for each program. LogicSigVersion >= 4. |
```

Comment:
It's too hard to make the sizes of the two programs independent? I've never written a ClearStateProgram more than ~10 bytes... it hurts me to pay for 1024 for both when I really only care about the ApprovalProgram

Reply:
Yes, we did both.  ExtraProgramPages gives you more space.  The changes to MaxAppProgramLen and introduction of MaxAppTotalProgramLen make it so that the bytes are shared freely between the two programs.

Reply:
Excellent, I didn't catch the repercussions